utility.lua

```diff
@@ -1480,6 +1480,80 @@ function Auxiliary.MergedDelayEventCheck2(e,tp,eg,ep,ev,re,r,rp)
 		g:Clear()
 	end
 end
+--Once the card has been moved to the public area, it should be listened to again
+function Auxiliary.RegisterMergedDelayedEvent_ToSingelCard(c,code,event,g,event_code_singel)
+	local mt=getmetatable(c)
+	if not event_code_singel then 
+		event_code_singel=code+event+EVENT_CUSTOM 
```

Comment:
How do you make sure the `code+event` thing does not lead into a conflict?

I suggest we use bitwise operations to to generate this event code.

Reply:
However, it's not that necessary to generate another code for the actual event code. That's what the script should maintain.

In case when we should register multiple events, we could do as what it looked like before, as follows:

```lua
local g=Group.CreateGroup()
aux.RegisterMergedDelayedEvent_ToSingelCard(c,59071624,EVENT_TO_GRAVE,g)
aux.RegisterMergedDelayedEvent_ToSingelCard(c,59071624,EVENT_REMOVE,g)

-- use EVENT_CUSTOM+59071624 in the following effects
```

To make it even simpler, we could make it like:

```lua
aux.RegisterMergedDelayedEvent_ToSingelCard(c,59071624,{EVENT_TO_GRAVE,EVENT_REMOVE},g)
```

Reply:
update to solve